File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        // for(int i = 0; i < nums.size(); i++){
        //     cout << nums[i] << ' ' << endl;
        // }
        int liveCounter = 0;
        int trackedNum = nums[0];
        vector<int> output;
        for(int i = 0; i < nums.size(); i++){
            int currentNum = nums[i];
            if(trackedNum != currentNum){
                if(liveCounter >= k){
                    output.push_back(trackedNum);
                }
                trackedNum = currentNum;
                liveCounter = 1;
            }
            else{
                liveCounter++;
            }
        }
        if(liveCounter >= k){
            output.push_back(trackedNum);
        }
        return output;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        [1, 1, 2, 2]
        [], [1], [1,1], [2], [1,2], [1, 1, 2]
        def subsets(nums):
            if not nums:
                return ([[]], [[]])
            
            current_num = nums[-1]
            dup = False
            if len(nums) > 1:
                dup = (current_num == nums[-2])
            
            res = subsets(nums[:-1])
            ret = []
            if dup:
                for subset in res[1]:
                    new_arr = subset[:]
                    new_arr.append(current_num)
                    ret.append(new_arr)
            else:
                for subset in res[0]:
                    new_arr = subset[:]
                    new_arr.append(current_num)
                    ret.append(new_arr)
            return (res[0]+ret, ret)
        final = subsets(nums)
        return final[0]
        
        
